MetaMask/metamask-extension
# Add Blockaid / PPOM Failed Request Security Alert

## Explanation


Fixes: https://github.com/MetaMask/MetaMask-planning/issues/1056

Before these changes, we displayed no UI for failed PPOM responses.

This PR adds supporting UI for failed PPOM responses. It optionally hides SecurityProviderBannerAlert footer.

I also snuck in a 1 line fix to fix the alignment of the footer items [8300833](https://github.com/MetaMask/metamask-extension/pull/20362/commits/8300833532dfc8169408f0f93c9e3127f42d4341)

## Screenshots/Screencaps



### Before

Before, no alert was shown


### After

<img width="1076" alt="Screenshot 2023-08-03 at 6 49 45 PM" src="https://github.com/MetaMask/metamask-extension/assets/20778143/c845c5ea-fbd1-43bc-a55e-be54151d7316">


## Manual Testing Steps

1. Run storybook: yarn storybook
2. See BlockaidBannerAlert page
3. Change reason value to "Failed"
4. Observe UI


## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
